lib/r10k/action/deploy/environment.rb

```diff
@@ -43,6 +44,30 @@ def call
 
         private
 
+        def extract_credentials
+          if @sshkey_path && @token_path
+            raise R10K::Error, "Cannot specify both an SSH key and a token to use with this deploy."
+          end
+
+          if @sshkey_path
+            path = convert_dash_to_stdin(@sshkey_path)
+            if File.exists?(path)
+              # TODO if this is reading from stdin, write the key to a (password
+              # protected?) tempfile instead and save the path for passing down
+              @settings[:sshkey] = File.read(path)
+            end
+          elsif @token_path
+            path = convert_dash_to_stdin(@sshkey_path)
+            if File.exists?(path)
+              @settings[:token] = File.read(path)
+            end
+          end
```

Comment:
Do we want to do validation of the file contents at this level?

Reply:
I'll see if there is a spec for OAuth token format that we can rely on.

Reply:
For reference, can I see where you found what you did?

Reply:
https://www.oauth.com/oauth2-servers/access-tokens/access-token-response/

Reply:
thanks!